msticpy/data/drivers/sumologic_driver.py

```diff
@@ -0,0 +1,481 @@
+        cs_dict: Dict[str, Any] = self._CONNECT_DEFAULTS
+        # Fetch any config settings
+        cs_dict.update(self._get_config_settings())
+        # If a connection string - parse this and add to config
+        if kwargs:
+            # if connection args supplied as kwargs
+            cs_dict.update(kwargs)
+            check_kwargs(cs_dict, list(SUMOLOGIC_CONNECT_ARGS.keys()))
+
+        missing_args = set(self._SUMOLOGIC_REQD_ARGS) - cs_dict.keys()
+        if missing_args:
+            raise MsticpyUserConfigError(
+                "One or more connection parameters missing for Sumologic connector",
+                ", ".join(missing_args),
+                f"Required parameters are {', '.join(self._SUMOLOGIC_REQD_ARGS)}",
+                "All parameters:",
+                *[f"{arg}: {desc}" for arg, desc in SUMOLOGIC_CONNECT_ARGS.items()],
+                title="no Sumologic connection parameters",
+            )
+        return cs_dict
+
+    def _query(
+        self, query: str, query_source: QuerySource = None, **kwargs
+    ) -> Union[pd.DataFrame, Any]:
+        """
+        Execute Sumologic query and retrieve results.
+
+        Parameters
+        ----------
+        query : str
+            Sumologic query to execute
+
+        Other Parameters
+        ----------------
+        kwargs :
+            Are passed to Sumologic
+            days: Search the past X days.
+            start_time: A datetime() object representing the start of the search
+                    window. If used without end_time, the end of the search
+                    window is the current time.
+            end_time: A datetime() object representing the end of the search window.
+                  If used without start_time, the search start will be the earliest
+                  time in the index.
+            timeZone: timezone used for time range search
+            byReceiptTime:  if time reference should used _receiptTime (time when Sumologic
+                        got message) instead of _messageTime (time present in log message).
+            limit: An integer describing the max number of search results to return.
+            forceMessagesResults: Force results to be raw messages even if aggregated query.
+            verbosity: Provide more verbose state. from 0 least verbose to 4 most one.
+            timeout: timeout when gathering results
+
+        Returns
+        -------
+        Union[pd.DataFrame, Any]
+            Query results in a dataframe.
+            or query response if an error.
+
+        """
+        del query_source
+        if not self._connected:
+            raise self._create_not_connected_err()
+
+        verbosity = kwargs.pop("verbosity", 0)
+        timeZone = kwargs.pop("timeZone", 'UTC')
+        byReceiptTime = kwargs.pop("byReceiptTime", False)
+        forceMessagesResults = kwargs.pop("forceMessagesResults", False)
+        timeout = kwargs.pop("timeout", 300)
```

Comment:
I don't see timeout used anywhere in this function - self.timeout is referenced however.

Reply:
fixed. self.timeout